### DISCUSSION TITLE
Publish Event and Sample Event

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5614] [Q&A]
hello elsa,

can you give some json example worflow that using publish event and event in elsa 3 ?

**Comment by glime-ai** [comment]
Certainly! Below is an example of a JSON workflow definition that uses a Publish Event and an Event activity in Elsa 3. This example assumes you have a basic understanding of how to define workflows in JSON format with Elsa.

```json
{
  "name": "PublishAndHandleEventWorkflow",
  "displayName": "Publish and Handle Event Workflow",
  "version": 1,
  "activities": [
    {
      "activityId": "publish-event-activity",
      "type": "PublishEvent",
      "properties": {
        "EventName": {
          "syntax": "Literal",
          "expression": "MyCustomEvent"
        },
        "EventData": {
          "syntax": "Literal",
          "expression": "{\"Message\":\"Hello World\"}"
        }
      }
    },
    {
      "activityId": "event-activity",
      "type": "Event",
      "properties": {
        "EventName": {
          "syntax": "Literal",
          "expression": "MyCustomEvent"
        }
      }
    },
    {
      "activityId": "write-line-activity",
      "type": "WriteLine",
      "properties": {
        "Text": {
          "syntax": "Literal",
          "expression": "Event received!"
        }
      }
    }
  ],
  "connections": [
    {
      "sourceActivityId": "publish-event-activity",
      "targetActivityId": "event-activity",
      "outcome": "Done"
    },
    {
      "sourceActivityId": "event-activity",
      "targetActivityId": "write-line-activity",
      "outcome": "Triggered"
    }
  ]
}
```

In this workflow:

- The `PublishEvent` activity publishes an event named `MyCustomEvent` with some JSON data.
- The `Event` activity listens for an event with the same name `MyCustomEvent`.
- When the event is triggered, the workflow continues to the `WriteLine` activity, which writes a message to the output.

Please note that the actual property names and values may vary depending on the version of Elsa you are using and the specific configuration of your activities. For more detailed instructions and examples, you should refer to the official Elsa documentation and the guides available on the Elsa Documentation Website.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
